parquet-common/src/main/java/org/apache/parquet/bytes/BytesInput.java

```diff
@@ -376,7 +377,11 @@ void writeInto(ByteBuffer buffer) {
         ByteBuffer workBuf = buffer.duplicate();
         int pos = buffer.position();
         workBuf.limit(pos + byteCount);
-        Channels.newChannel(in).read(workBuf);
+        ReadableByteChannel channel = Channels.newChannel(in);
+        int remaining = byteCount;
+        while (remaining > 0) {
+          remaining -= channel.read(workBuf);
```

Comment:
Is `remaining` reliable? Should we check the return value of `channel.read(workBuf)`?

Reply:
added a check to detect EOF case